brave/src/main/java/brave/CurrentSpanCustomizer.java

```diff
@@ -0,0 +1,56 @@
+package brave;
+
+/**
+ * Provides a mechanism for end users to be able to customise the current span.
+ *
+ * <p>Handles the case of there being no current span in scope.
+ */
+public final class CurrentSpanCustomizer implements SpanCustomizer {
```

Comment:
@reta @jplock @llinder @devinsba FYI, this is a way to prevent users from having or needing a reference to the tracer. For example, provide `SpanCustomizer` via this type. Then if users employ `SpanCustomizer`, they can contribute to a current span w/o using any advanced types.